api_client/python/timesketch_api_client/client.py

```diff
@@ -637,6 +637,47 @@ def explore(self,
 
         return response_json
 
+    def run_analyzer(
+            self, analyzer_name, analyzer_kwargs=None, timeline_id=None,
+            timeline_name=None):
+        """Run an analyzer on a timeline.
+
+        Args:
+            analyzer_name: the name of the analyzer class to run against the
+                timeline.
+            analyzer_kwargs: optional dict with parameters for the analyzer.
+                This is optional and just for those analyzers that can accept
+                further parameters.
+            timeline_id: the data store index for the timeline. This is optional
+                and only required if timeline_name is not set.
+            timeline_name: the name of the timeline in the timesketch UI. This
+                is optional and only required if timeline_id is not set.
+
+        Returns:
+            A string with the results of the API call to run the analyzer.
+        """
+        if not timeline_id and not timeline_name:
```

Comment:
This is risky, there can be multiple timelines with the same name in a sketch. We don't enforce unique names today. I say that you need to specify the ID here to be on the safe side.

Reply:
changed this a bit... removed the entire notion of timeline_name from the actual API and here I'm going to check if you use timeline name and there is only one result, then I'll proceed.